File: aexample/src/main/java/org/judal/examples/java/jdbc/E28_BlobReadWrite.java
package org.judal.examples.java.jdbc;

import org.junit.Test;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertArrayEquals;

import java.io.InputStream;

import org.judal.examples.Resources;
import org.judal.examples.java.model.Student;
import org.judal.storage.java.TableOperation;

/**
 * Example of how to write and read a LONGVARBINARY field
 */
public class E28_BlobReadWrite {

	@Test
	public void demo() throws Exception {
		
		final int photoSize = 4832;
		byte[] photoBytes = new byte[photoSize];
		
		setUp();

		// Blobs are written and read using byte arrays as any other column
		try (TableOperation<Student> op = new TableOperation<>(new Student())) {
			Student s1 = op.load(new Integer(1));
			
			try (InputStream photo = Resources.getResourceAsStream("photo.jpg")) {
				assertNotNull(photo);
				photo.read(photoBytes);				
			}
			s1.put("photo", photoBytes);
			s1.store();			
		}

		try (TableOperation<Student> op = new TableOperation<>(new Student())) {
			Student s1 = op.load(new Integer(1));
			assertArrayEquals(photoBytes, s1.getBytes("photo"));
		}

		tearDown();
	}

	public static void setUp() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.setUp();
		E10_WriteCSVDataIntoTheDatabase.insertStudentsIntoDatabase();
	}

	public static void tearDown() throws Exception {
		E10_WriteCSVDataIntoTheDatabase.tearDown();
	}
	
}

File: aexample/src/main/scala/org/judal/examples/scala/jdbc/E28_BlobReadWrite.scala
package org.judal.examples.scala.jdbc

import org.junit.Test
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertArrayEquals

import java.io.InputStream

import org.judal.examples.Resources
import org.judal.examples.java.model.map.Student
import org.judal.storage.java.TableOperation

import org.judal.Using._

/**
 * Example of how to write and read a LONGVARBINARY field
 */
class E28_BlobReadWrite {

	@Test
	def demo() : Unit = {
		
		val photoSize = 4832
		val photoBytes = new Array[Byte](photoSize)
		
		E28_BlobReadWrite.setUp()
		var op : TableOperation[Student] = null
		using (op) {
		  // Blobs are written and read using byte arrays as any other column
		  op = new TableOperation[Student](new Student())
			val s1 = op.load(new Integer(1))
			var photo : InputStream  = null
			using (photo) {
			  photo = Resources.getResourceAsStream("photo.jpg")
				assertNotNull(photo)
				photo.read(photoBytes)				
			}
			s1.put("photo", photoBytes)
			s1.store()			
		}

		using (op) {
		  op = new TableOperation[Student](new Student())
			val s1 = op.load(new Integer(1))
			assertArrayEquals(photoBytes, s1.getBytes("photo"))
		}

		E28_BlobReadWrite.tearDown()
	}
	
}


object E28_BlobReadWrite {
  
	def setUp() : Unit = {
		E10_WriteCSVDataIntoTheDatabase.setUp()
		E10_WriteCSVDataIntoTheDatabase.insertStudentsIntoDatabase()
	}

	def tearDown() : Unit = {
		E10_WriteCSVDataIntoTheDatabase.tearDown()
	}

}